templates/openwisp2/settings.py

```diff
@@ -494,3 +494,37 @@
 {{ instruction }}
 
 {% endfor %}
+
+{% if django_cors.enabled %}
+
+from corsheaders.defaults import default_methods, default_headers
+
+INSTALLED_APPS.append('corsheaders')
+MIDDLEWARE.insert(MIDDLEWARE.index('django.middleware.common.CommonMiddleware'), 'corsheaders.middleware.CorsMiddleware')
+{% if django_cors.replace_https_referer %}
+MIDDLEWARE.insert(MIDDLEWARE.index('django.middleware.csrf.CsrfViewMiddleware') + 1, 'corsheaders.middleware.CorsPostCsrfMiddleware')
```

Comment:
**+1** because `'corsheaders.middleware.CorsPostCsrfMiddleware'` must be placed after `'django.middleware.csrf.CsrfViewMiddleware'`